pkg/scalers/predictkube_scaler.go

```diff
@@ -0,0 +1,463 @@
+			}
+
+			s, err := strconv.ParseFloat(res.Value, 64)
+			if err != nil {
+				return nil, err
+			}
+
+			out = append(out, &commonproto.Item{
+				Timestamp:  t,
+				Value:      s,
+				MetricName: metricName,
+			})
+		}
+	}
+
+	return out, nil
+}
+
+func parsePredictKubeMetadata(config *ScalerConfig) (result *predictKubeMetadata, err error) {
+	validate := validator.New()
+	meta := predictKubeMetadata{}
+
+	if val, ok := config.TriggerMetadata["query"]; ok {
+		if len(val) == 0 {
+			return nil, fmt.Errorf("no query given")
+		}
+
+		meta.query = val
+	} else {
+		return nil, fmt.Errorf("no query given")
+	}
+
+	if val, ok := config.TriggerMetadata["prometheusAddress"]; ok {
+		err = validate.Var(val, "url")
+		if err != nil {
+			return nil, fmt.Errorf("invalid prometheusAddress")
+		}
+
+		meta.prometheusAddress = val
+	} else {
+		return nil, fmt.Errorf("no prometheusAddress given")
+	}
+
+	if val, ok := config.TriggerMetadata["predictHorizon"]; ok {
+		meta.predictHorizon, err = str2duration.ParseDuration(val)
+		if err != nil {
+			return nil, fmt.Errorf("predictHorizon parsing error %s", err.Error())
+		}
+	} else {
+		return nil, fmt.Errorf("no predictHorizon given")
+	}
+
+	if val, ok := config.TriggerMetadata["queryStep"]; ok {
+		meta.stepDuration, err = str2duration.ParseDuration(val)
+		if err != nil {
+			return nil, fmt.Errorf("queryStep parsing error %s", err.Error())
+		}
+	} else {
+		return nil, fmt.Errorf("no queryStep given")
+	}
+
+	if val, ok := config.TriggerMetadata["historyTimeWindow"]; ok {
+		meta.historyTimeWindow, err = str2duration.ParseDuration(val)
+		if err != nil {
+			return nil, fmt.Errorf("historyTimeWindow parsing error %s", err.Error())
+		}
+	} else {
+		return nil, fmt.Errorf("no historyTimeWindow given")
+	}
+
+	if val, ok := config.TriggerMetadata["threshold"]; ok {
+		meta.threshold, err = strconv.ParseInt(val, 10, 64)
+		if err != nil {
+			return nil, fmt.Errorf("threshold parsing error %s", err.Error())
+		}
+	} else {
+		return nil, fmt.Errorf("no threshold given")
+	}
+
+	if val, ok := config.TriggerMetadata["metricName"]; ok {
+		if len(val) == 0 {
+			return nil, fmt.Errorf("no metric name given")
+		}
+
+		err = validate.Var(val, "ascii")
+		if err != nil {
+			return nil, fmt.Errorf("invalid apiKey")
+		}
+
+		meta.metricName = val
+	} else {
+		return nil, fmt.Errorf("no metric name given")
+	}
+
+	meta.scalerIndex = config.ScalerIndex
+
+	if val, ok := config.AuthParams["apiKey"]; ok {
+		err = validate.Var(val, "jwt")
+		if err != nil {
+			return nil, fmt.Errorf("invalid apiKey")
+		}
+
+		meta.apiKey = val
+	} else {
+		return nil, fmt.Errorf("no api key given")
+	}
+
+	return &meta, nil
+}
+
+func (pks *PredictKubeScaler) ping(ctx context.Context) (err error) {
+	_, err = pks.api.Runtimeinfo(ctx)
+	return err
+}
+
+// initPredictKubePrometheusConn init prometheus client and setup connection to API
+func (pks *PredictKubeScaler) initPredictKubePrometheusConn() (err error) {
```

Comment:
Please, receive the `context` as parameter and pass it to `ping` function. There is a context available when this function is called and it gives the option to KEDA for stopping it (`NewPredictKubeScaler`)

Reply:
done

Reply:
done